File: Java/src/Solution11.java
public class Solution11 {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int area = (right - left) * (height[left] < height[right] ? height[left] : height[right]);
        while (left < right) {
            if (height[left] < height[right]) {
                int i = left;
                while (i < right && height[i] <= height[left]) {
                    i++;
                }
                if (i != right) {
                    left = i;
                    if ((right - left) * (height[left] < height[right] ? height[left] : height[right]) > area) {
                        area = (right - left) * (height[left] < height[right] ? height[left] : height[right]);
                    }
                } else {
                    break;
                }
            } else {
                int i = right;
                while (i > left && height[i] <= height[right]) {
                    i--;
                }
                if (i != left) {
                    right = i;
                    if ((right - left) * (height[left] < height[right] ? height[left] : height[right]) > area) {
                        area = (right - left) * (height[left] < height[right] ? height[left] : height[right]);
                    }
                } else {
                    break;
                }
            }
        }
        return area;
    }

    public static void main(String[] args) {
        Solution11 solution11 = new Solution11();
        int[] nums = {1,1};
        int result = solution11.maxArea(nums);
        System.out.println(result);
    }
}

File: C++/src/Solution1_100/Solution11.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution11.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

//int Solution::maxArea(vector<int> &height) {
//    int n = height.size();
//    int left = 0, right = n - 1;
//    int total = min(height[left], height[right]) * (right - left);
//    while(left < right) {
//        if (height[left] >= height[right]) {
//            right--;
//        }else if (height[left] < height[right]){
//            left++;
//        }
//        total = max(total, min(height[left], height[right]) * (right - left));
//    }
//    return total;
//}


int maxArea(vector<int>& height) {
  int n = height.size();
  int left = 0, right = n - 1;
  int res = 0;
  while (left < right){
    res = max(res, min(height[left], height[right]) * (right - left));
    if (height[left] < height[right]){
      left++;
    }else{
      right--;
    }
  }
  res = max(res, min(height[left], height[right]) * (right - left));
  return res;
}


